File: src/main/kotlin/net/chrissearle/sbanken/model/Transaction.kt
package net.chrissearle.sbanken.model

data class Transaction(
    val accountingDate: String,
    val interestDate: String?,
    val otherAccountNumber: String?,
    val otherAccountNumberSpecified: Boolean,
    val amount: Double,
    val text: String?,
    val transactionType: String?,
    val transactionTypeCode: Int,
    val transactionTypeText: String,
    val isReservation: Boolean,
    val reservationType: ReservationType?,
    val source: SourceType,
    val cardDetailsSpecified: Boolean,
    val cardDetails: CardDetails?,
    val transactionDetailSpecified: Boolean,
    val transactionDetail: TransactionDetail?
)

File: vue/src/types/Transaction.ts
interface CardDetails {
  cardNumber: string | null;
  currencyAmount: number;
  currencyRate: number;
  merchantCategoryCode: string | null;
  merchantCategoryDescription: string | null;
  merchantCity: string | null;
  merchantName: string | null;
  originalCurrencyCode: string | null;
  purchaseDate: string;
  transactionId: string;
}

interface TransactionDetail {
  formattedAccountNumber: string | null;
  transactionId: number;
  cid: string;
  amountDescription: string;
  receiverName: string | null;
  numericReference: number;
  payerName: string | null;
  registrationDate: string | null;
}

export default interface Transaction {
  accountingDate: string;
  interestDate: string | null;
  otherAccountNumber: string | null;
  otherAccountNumberSpecified: boolean;
  amount: number;
  text: string | null;
  transactionType: string | null;
  transactionTypeCode: number;
  transactionTypeText: string;
  isReservation: boolean;
  reservationType: string | null;
  source: string | null;
  cardDetailsSpecified: boolean;
  cardDetails: CardDetails | null;
  transactionDetailSpecified: boolean;
  transactionDetail: TransactionDetail | null;
}
